src/eduid/satosa/scimapi/accr.py

```diff
@@ -0,0 +1,148 @@
+            "internal_accr_rewrite_map"
+        )
+
+        super().__init__(*args, **kwargs)
+
+    def process(self, context: satosa.context.Context, data: satosa.internal.InternalData) -> ProcessReturnType:
+        requested_accr: list = context.get_decoration(Context.KEY_AUTHN_CONTEXT_CLASS_REF)
+
+        if requested_accr:
+            supported_accr_to_forward = []
+            for accr in requested_accr:
+                if accr in self.supported_accr_sorted_by_prio:
+                    supported_accr_to_forward.append(accr)
+                else:
+                    logger.debug(f"Removing unsupported ACCR ({accr}) from request")
+
+            if not supported_accr_to_forward:
+                raise SATOSAAuthenticationError(context.state, "Unsupported ACCR")
+            else:
+                requested_accr = supported_accr_to_forward
+
+        internal_accr_rewrite_map = self.internal_accr_rewrite_map
+        if internal_accr_rewrite_map and requested_accr:
+            context.state["internal_accr_rewrite_map"] = internal_accr_rewrite_map
+            for index, value in enumerate(requested_accr):
+                if value in internal_accr_rewrite_map:
+                    logger.debug(f"Remapping ACCR for internal use. From {value} to {internal_accr_rewrite_map[value]}")
+                    requested_accr[index] = internal_accr_rewrite_map[value]
+
+        context.state["saved_accr"] = requested_accr
+        context.state["supported_accr_sorted_by_prio"] = self.supported_accr_sorted_by_prio
+        accr_to_forward = requested_accr
+        logger.debug(f"Saving requested ACCR for later use: {requested_accr}).")
+
+        virtual_idp = context.target_frontend
+        if self.lowest_accepted_accr_for_virtual_idp:
+            minimum_accr = self.lowest_accepted_accr_for_virtual_idp.get(virtual_idp)
+        if minimum_accr:
+            logger.debug(f"Minimum accepted ACCR for {virtual_idp} is: {minimum_accr}.")
+            required_accr_by_virtual_idp, _ = self.split_sorted_accr_by(
+                minimum_accr=minimum_accr, supported_accr=self.supported_accr_sorted_by_prio
+            )
+            logger.debug(
+                f"Replacing requested ACCR: {requested_accr}, with what {virtual_idp} requires: {required_accr_by_virtual_idp}."
+            )
+            accr_to_forward = required_accr_by_virtual_idp
+
+        context.state[Context.KEY_TARGET_AUTHN_CONTEXT_CLASS_REF] = accr_to_forward
+
+        return super().process(context, data)
+
+    @staticmethod
+    def split_sorted_accr_by(minimum_accr: str, supported_accr: list) -> tuple[list, list]:
+        result = []
+        temp_list = []
+        for i in supported_accr:
+            if i == minimum_accr:
+                temp_list.append(i)
+                result.append(temp_list)
+                temp_list = []
+            else:
+                temp_list.append(i)
+        result.append(temp_list)
+        return result[0], result[1]
```

Comment:
I think this will yield the same result in a more clear way
```
minimum_accr = "second_accr"
supported_accr = ["first_accr", "second_accr", "third_accr"]

if minimum_accr in supported_accr:
    best_to_minimum = supported_accr[:supported_accr.index(minumum_accr)+1]
    rest = supported_accr[supported_accr.index(minumum_accr)+1:]
```

Reply:
Yes! Thanks.